apps/prairielearn/src/models/enrollment.ts

```diff
@@ -21,24 +24,46 @@ import { isEnterprise } from '../lib/license.js';
 import { HttpRedirect } from '../lib/redirect.js';
 import { assertNever } from '../lib/types.js';
 
+import { type SupportedActionsForTable, insertAuditEvent } from './audit-event.js';
 import { generateUsers, selectUserById } from './user.js';
 
 const sql = loadSqlEquiv(import.meta.url);
 
-export async function enrollInvitedUserInCourseInstance({
-  course_instance_id,
+async function dangerouslyEnrollInvitedUserInCourseInstance({
+  enrollment_id,
   user_id,
-  pending_uid,
+  agent_user_id,
+  agent_authn_user_id,
+  action_detail,
 }: {
-  course_instance_id: string;
+  enrollment_id: string;
   user_id: string;
-  pending_uid: string;
-}): Promise<void> {
-  await execute(sql.enroll_invited_user_in_course_instance, {
-    course_instance_id,
-    pending_uid,
-    user_id,
+  agent_user_id: string | null;
+  agent_authn_user_id: string | null;
+  action_detail: SupportedActionsForTable<'enrollments'>;
+}): Promise<Enrollment> {
+  const oldEnrollment = await selectEnrollmentById({ id: enrollment_id });
+  const newEnrollment = await queryRow(
+    sql.enroll_invited_user,
```

Comment:
The `WHERE` clause in this query means this could return zero rows. Is the "dangerously" here meant to represent that this should only be called with the existing state already having been validated? If so, is the `WHERE` clause needed?

Consider a JSDoc comment.

Reply:
I added a comment, and reworked the WHERE to an assert in 500882160706cd6f78fce69238ab2fbabbd74964.